File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetCreateResponse.java
package com.luckyseven.backend.domain.budget.dto;

import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Builder;
import lombok.Getter;

public record BudgetCreateResponse (
    @NotNull Long id,
    @NotNull LocalDateTime createdAt,

    @NotNull Long setBy,
    @NotNull BigDecimal balance,

    @NotNull BigDecimal avgExchangeRate,
    @NotNull BigDecimal foreignBalance
) { }

File: Backend/src/main/java/com/luckyseven/backend/domain/budget/dto/BudgetCreateResponse.kt
package com.luckyseven.backend.domain.budget.dto

import jakarta.validation.constraints.NotNull
import java.math.BigDecimal
import java.time.LocalDateTime

data class BudgetCreateResponse(

    val id: Long,
    val createdAt: LocalDateTime,

    val setBy: Long,
    val balance: BigDecimal,

    val avgExchangeRate: BigDecimal,
    val foreignBalance: BigDecimal

)
